cpp/arcticdb/storage/s3/nfs_backed_storage.cpp

```diff
@@ -208,16 +208,22 @@ static std::string prefix_handler(const std::string& prefix, const std::string&
 }
 
 bool NfsBackedStorage::do_iterate_type_until_match(KeyType key_type, const IterateTypePredicate& visitor, const std::string& prefix) {
+    auto iter_prefix_handler = [] ([[maybe_unused]] const std::string& prefix, const std::string& key_type_dir, [[maybe_unused]] const KeyDescriptor& key_descriptor, [[maybe_unused]] KeyType) {
+        // The prefix handler is not used for filtering (done in func below)
+        // so we just return the key type dir
+        return key_type_dir;
+    };
+    
     const IterateTypePredicate func = [&v = visitor, prefix=prefix] (VariantKey&& k) {
         auto key = unencode_object_id(k);
-        if(prefix.empty() || variant_key_id(key) == StreamId{prefix}) {
+        if(prefix.empty() || variant_key_view(key).find(prefix) != std::string::npos) {
```

Comment:
I think a comment here would be helpful as people often forget that we're handling these path's differently.